vault/expiration.go

```diff
@@ -214,9 +236,26 @@ func (r *revocationJob) OnFailure(err error) {
 		return
 	}
 
+	if errIsUnrecoverable(err) {
```

Comment:
Could we move this below and make it:

`if pending.revokesAttempted >= maxRevokeAttempts || errIsUnrecoverable(err) {`

Reply:
we can, but then we'll need logic to decide which error to pass to `markLeaseAsZombie()`

Reply:
i've got an idea for a clean combo, i'll comment the commit when done

Reply:
6f82bb338f